### ISSUE TITLE
Appender search path

### ISSUE THREAD

**Issue by Enerccio** [nomiddlename/log4js-node#368]
So far, it is rather stupidly handled, it searches for whatever string I put or in its own appenders, but that would require me to put new appenders into log4js directory in node_modules, which is stupid. There should be a way to provide local path to the project or something.

**Comment by nomiddlename** [comment]
I see the problem. I think at some point in the distant past `require` did something slightly different to what it does now. Now you need to specify the appender name as a path relative to the log4js lib directory inside node_modules. It's probably easier to make your appender a package in its own right, and specify it as a dependency in your package.json, and then refer to it in your config with just the package name rather than a path. I don't think there's a way for log4js to load a module from your project easily any more.

**Comment by Enerccio** [comment]
The best way would be for me to simply provide 'appender', initfunction instead, eliminating the need for require at all (which I can do at runtime, but not from configuration file)

**Comment by babadofar** [comment]
I had problems using my own custom appender as well, due to to the same path quirks you mention above. FYI, the code below is one way to use your own home grown appender 

```
const  eventhubAppender = require('./modules/logging/log4js/eventhub_appender');
log4js.loadAppender('eventHubAppender', eventhubAppender);
log4js.addAppender(log4js.appenders['eventHubAppender'](config));
 ```

**Comment by nomiddlename** [comment]
Going to fix this by searching for appenders by using `process.cwd` in the call to require. This should make things a bit more intuitive. Note that `loadAppender` (or `addAppender`) does not exist in version 2.x.

**Comment by nomiddlename** [comment]
Published to nom in version 2.2.0
